frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx

```diff
@@ -0,0 +1,330 @@
+          value: lblVal,
+          path: labelPath,
+        },
+      ];
+      return k8sPatch(NodeModel, node, patch);
+    });
+    return labelData;
+  };
+
+  // tainting the selected nodes
+  const taintNodes = (selectedNode: NodeKind[]) => {
+    const taintData = selectedNode
+      .filter((node: NodeKind) => {
+        const roles = getNodeRoles(node);
+        // don't taint master nodes as its already tainted
+        return roles.indexOf('master') === -1;
+      })
+      .map((node) => {
+        const taints = node.spec && node.spec.taints ? [...node.spec.taints, taintObj] : [taintObj];
+        const patch = [
+          {
+            value: taints,
+            path: '/spec/taints',
+            op: node.spec.taints ? 'replace' : 'add',
+          },
+        ];
+        return k8sPatch(NodeModel, node, patch);
+      });
+
+    return taintData;
+  };
+
+  const submit = (event) => {
+    event.preventDefault();
+    setProgress(true);
+    setError('');
+
+    const selectedData: NodeKind[] = _.filter(nodes, 'selected');
+    const promises = [];
+
+    promises.push(...labelNodes(selectedData));
+    promises.push(...taintNodes(selectedData));
+
+    const obj = _.cloneDeep(ocsRequestData);
+    obj.spec.dataDeviceSet.storageClassName = storageClass;
+    promises.push(k8sCreate(OCSServiceModel, obj));
+
+    Promise.all(promises)
+      .then(() => {
+        history.push(
+          `/k8s/ns/${ocsProps.namespace}/clusterserviceversions/${
+            ocsProps.clusterServiceVersion.metadata.name
+          }/${referenceForModel(OCSServiceModel)}/${obj.metadata.name}`,
+        );
+        setProgress(false);
+        setError('');
+      })
+      .catch((err) => {
+        setProgress(false);
+        setError(err.message);
+      });
+  };
+
+  return (
+    <>
+      <Table
+        onSelect={onSelect}
+        cells={columns}
+        rows={nodes}
+        sortBy={sortBy}
+        onSort={onSort}
+        variant={TableVariant.compact}
+      >
+        <TableHeader />
+        <TableBody />
+      </Table>
+      <p className="control-label help-block" id="nodes-selected">
+        {selectedNodesCnt} node(s) selected
+      </p>
+      <div className="form-group">
+        <Tooltip content="The Storage Class will be used to request storage from the underlying infrastructure to create the backing persistent volumes that will be used to provide the OpenShift Container Storage (OCS) service">
+          <OCSStorageClassDropdown
+            onChange={handleStorageClass}
+            id="storageclass-dropdown"
+            required={false}
+            name="storageClass"
+          />
+        </Tooltip>
+      </div>
+      <ButtonBar errorMessage={error} inProgress={inProgress}>
+        <ActionGroup className="pf-c-form">
+          <Button
+            type="button"
+            variant="primary"
+            onClick={submit}
+            isDisabled={selectedNodesCnt < minSelectedNode}
```

Comment:
Do we have a message somewhere saying you need to select at least 3 nodes? Otherwise it's not going to be obvious why the button is disabled.

cc @julienlim

Reply:
@spadgett yes. its there at the top of the page. Attaching the screenshot - 
![Screenshot from 2019-08-23 10-44-40](https://user-images.githubusercontent.com/5517376/63566390-e6914080-c5a7-11e9-9734-e7130438585d.png)